File: python/70_Climbing_Stairs.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def climbStairs(self, n):
        if n==1:
            return 1
        elif n==2:
            return 2
        else:
            a,b,i=1,2,3
            while i<=n:
                a,b,i=b,a+b,i+1
            return b

File: Java/70_Climbing_Stairs.java
public class Solution {
    public int climbStairs(int n) {
        // 斐波那契数列
        if(n<=1) return 1;// 小于等于1则返回1
        int a=1,b=2;
        while(n>2){
            b=b+a;
            a=b-a;
            n--;
        }
        return b;// 对于返回类型int来说，n超过一定限度会溢出，这种情况leetcode的测例并没有考虑
    }
}